src/cohort-builder.resource.ts

```diff
@@ -25,24 +26,23 @@ export const search = async (searchParams: SearchParams) => {
 /**
  * @returns Locations
  */
-export const fetchLocations = async () => {
-  const locationsResp: FetchResponse<{
-    results: Response[];
-  }> = await openmrsFetch("/ws/rest/v1/location", {
-    method: "GET",
-  });
+export const useLocations = () => {
+  const { data, error } = useSWRImmutable<{
+    data: { results: Response[] };
+  }>("/ws/rest/v1/location", openmrsFetch);
 
-  const {
-    data: { results },
-  } = locationsResp;
   const locations: DropdownValue[] = [];
-  results.map((location: Response, index: number) => {
+  data?.data.results.map((location: Response, index: number) => {
     locations.push({
       id: index,
       label: location.display,
       value: location.uuid,
     });
   });
 
-  return locations;
+  return {
+    isLoading: !data && !error,
+    locations,
+    locationsError: error,
```

Comment:
Normally you'd just call this `error`, and let the users of the hook resolve any naming conflicts. I leave it at your discretion.